api.go

```diff
@@ -591,16 +591,16 @@ func (c Client) executeMethod(ctx context.Context, method string, metadata reque
 		}
 	}
 
-	// Create a done channel to control 'newRetryTimer' go routine.
-	doneCh := make(chan struct{}, 1)
+	// Create cancel context to control 'newRetryTimer' go routine.
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
One last thing. I think it is cleaner if you call the new variable retryCtx so it won't interfere with other contexts